docs/source/coroutines.rst

```diff
@@ -260,6 +260,180 @@ And of course, the sky is the limit when you compose the two.
     This means that Tasks passed to it are *still running*.
     You may need to cancel those Tasks with :meth:`.Task.cancel`.
 
+.. _task_manager_tutorial:
+
+:class:`!TaskManager`: Structured Asymmetric Concurrency
+========================================================
+
+The :class:`~cocotb.task.TaskManager` class is another way to run multiple async routines concurrently and wait for them all to complete,
+but properly manages the lifetime of its "children" and handles exceptions and cancellations gracefully.
+This should be the preferred way to run multiple coroutines concurrently.
+
+The typical usage pattern is to create a :class:`!TaskManager` as an :term:`asynchronous context manager` using the :keyword:`async with` statement;
+then to use the :deco:`~cocotb.triggers.TaskManager.fork` decorator method to start :class:`!Task`\ s within the context block.
```

Comment:
This somehow shows the hierarchy including the tilde in https://cocotb--5163.org.readthedocs.build/en/5163/coroutines.html#taskmanager-structured-asymmetric-concurrency